compiler/rustc_resolve/src/rustdoc.rs

```diff
@@ -498,20 +498,25 @@ pub fn span_of_fragments(fragments: &[DocFragment]) -> Option<Span> {
 /// This method does not always work, because markdown bytes don't necessarily match source bytes,
 /// like if escapes are used in the string. In this case, it returns `None`.
 ///
+/// `markdown` is typically the entire documentation for an item,
+/// after combining fragments.
+///
 /// This method will return `Some` only if:
 ///
 /// - The doc is made entirely from sugared doc comments, which cannot contain escapes
 /// - The doc is entirely from a single doc fragment, with a string literal, exactly equal
 /// - The doc comes from `include_str!`
+/// - The doc includes exactly one substring matching `markdown[md_range]` which is contained in a single doc fragment.
 pub fn source_span_for_markdown_range(
```

Comment:
Since #115689 this function resides in compiler, so Clippy can use it, too. You could add a comment explaining that but it's not super necessary since moving this function would lead to the Clippy subtree failing to build